### DISCUSSION TITLE
How can the client not interrupt when the server is down

### DISCUSSION THREAD

**Discussion by ccyy101** [open62541pp/open62541pp#199] [Q&A]
When I was running "open62541pp_examples_client_subscription" project, once the server application is shut down, it will cause the this project to crash.Can you help me solve this problem?

![1730d5294154d3b52c3f8c06d307157](https://github.com/open62541pp/open62541pp/assets/117834509/8584734e-7897-4475-9e4a-d1c11d0a797b)

**Comment by lukasberbuer** [comment]
Hi @ccyy101,

it seems to work fine on my machine (see video attached). Does the client crash with an exception?

https://github.com/open62541pp/open62541pp/assets/36054362/13c36f9a-f695-4177-a8e8-4609e1d613a2

**Reply by ccyy101** [reply]
Hi @lukasberbuer
thanks for your reply. I am using window+visual studio for compilation, this is the problem I encountered, once I shut down the server, the client will stop with an exception.(video, console information and report error in vs)

https://github.com/open62541pp/open62541pp/assets/117834509/bd1b3416-8410-4f1a-99f1-3ad16e92889c
![1708740361058](https://github.com/open62541pp/open62541pp/assets/117834509/e53086b7-2d97-4acd-9d88-8cf6d6a72a72)
![1708740625716](https://github.com/open62541pp/open62541pp/assets/117834509/dde9e424-7f37-4192-a837-c25cb9946ac0)

**Comment by lukasberbuer** [comment]
I could reproduce this on Windows. The problem is, that the status code returned if server disconnects is:
- `UA_STATUSCODE_BADDISCONNECT` on Linux
- `UA_STATUSCODE_BADTIMEOUT` on Windows

The loop tries to catch the `BadDisconnect` exception, which is not thrown on Windows. If you just use the base `BadStatus` exception, everything should work fine:

```cpp
    // Endless loop to automatically (try to) reconnect to server.
    while (true) {
        try {
            client.connect("opc.tcp://localhost:4840");
            // Run the client's main loop to process callbacks and events.
            // This will block until client.stop() is called or an exception is thrown.
            client.run();
        } catch (const opcua::BadDisconnect&) {  // <-- change this to BadStatus
            // Workaround to enforce a new session
            // https://github.com/open62541pp/open62541pp/issues/51
            client.disconnect();
            std::cout << "Disconnected. Retry to connect in 3 seconds\n";
            std::this_thread::sleep_for(std::chrono::seconds(3));
        }
    }
```

**Reply by lukasberbuer** [reply]
Fixed in #203.
